elements/howto-tabs/howto-tabs.js

```diff
@@ -47,54 +52,55 @@
 
       this.setAttribute('role', 'tablist');
 
-      // Before the elements starts booting, it waits for
-      // the both `<howto-tab>` and `<howto-tabpanel>` to load.
+      // For progressive enhancement, the markup should alternate between tabs
+      // and panels. Elements reordering children tend to not work well with
+      // frameworks. Instead ShadowDOM is used to reorder the elements by
+      // using slots.
+      this.shadowRoot.innerHTML = `
+        <slot name="tab"></slot>
```

Comment:
I've had prior conversations with the Polymer team where they said that having each instance call `innerHTML` incurs excess parsing costs. Back then they recommended a pattern like https://jsbin.com/jekacov/3/edit?html,output

I think you can ignore the `_mounted` stuff and the ShadyCSS bits. The main thing is reusing the template to cut out extra parse costs.